core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -13,86 +13,178 @@
         }
 
-        public Expression translate(ConnectorExpression expression)
+        @Override
+        protected Optional<Expression> visitFieldDereference(FieldDereference node, Void context)
+        {
+            Optional<Expression> base = process(node.getTarget());
+            return base.map(expression -> new SubscriptExpression(expression, new LongLiteral(Long.toString(node.getField() + 1))));
+        }
+
+        @Override
+        protected Optional<Expression> visitCall(Call node, Void context)
         {
-            if (expression instanceof Variable) {
-                return variableMappings.get(((Variable) expression).getName()).toSymbolReference();
+            if (LIKE_FUNCTION_NAME.equals(node.getName())) {
+                return visitLike(node.getArguments().get(0), node.getArguments().get(1), context);
             }
 
-            if (expression instanceof Constant) {
-                return literalEncoder.toExpression(((Constant) expression).getValue(), expression.getType());
+            QualifiedName qualifiedName = getQualifiedName(node);
+
+            List<Expression> arguments = node.getArguments().stream()
+                                             .map(argument -> ConnectorExpressionPredicateTranslator.toPredicate(argument, literalEncoder, metadata))
+                                             .collect(Collectors.toList());
+
+            if (REGEXP_LIKE_FUNCTION_NAME.equalsIgnoreCase(node.getName()) && arguments.size() == 2) {
+                Expression pattern = arguments.get(1);
+                if (pattern instanceof StringLiteral) {
+                    Slice slice = ((StringLiteral) pattern).getSlice();
+                    JoniRegexp joniRegexp = JoniRegexpCasts.castVarcharToJoniRegexp(slice);
+                    arguments.set(1, literalEncoder.toExpression(joniRegexp, JoniRegexpType.JONI_REGEXP));
+                }
             }
 
-            if (expression instanceof FieldDereference) {
-                FieldDereference dereference = (FieldDereference) expression;
-                return new SubscriptExpression(translate(dereference.getTarget()), new LongLiteral(Long.toString(dereference.getField() + 1)));
+            return Optional.of(new FunctionCall(Optional.empty(),
+                                                qualifiedName,
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                false,
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                arguments));
+        }
+
+        protected Optional<Expression> visitLike(ConnectorExpression value, ConnectorExpression pattern, Void context)
```

Comment:
This is not part of the visitor interface, so it shouldn't be called `visitLike`. Rename it to `translateLike`.

Also, the `context` argument is not used or needed.

Reply:
Done